interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+    def __init__(self, name='', elements='', species='', note='',
+                 reactions='none', site_density=0.0, phases=[],
+                 kinetics='Interface', transport='None', initial_state=None,
+                 options=[]):
+        """
+        The parameters correspond to those of :class:`.phase`, with the
+        following modifications:
+
+        :param reactions:
+            The heterogeneous reactions at this interface. If omitted, no
+            reactions will be included. A string or sequence of strings in the
+            format described in `Declaring the Reactions
+            <https://cantera.org/tutorials/cti/phases.html#declaring-the-reactions>`_.
+        :param site_density:
+            The number of adsorption sites per unit area.
+        :param phases:
+            A string listing the bulk phases that participate in reactions
+            at this interface.
+        """
+        phase.__init__(self, name, 2, elements, species, note, reactions,
+                       initial_state, options)
+        self.thermo_model = 'ideal-surface'
+        self._kin = kinetics
+        self._tr = None if transport == 'None' else transport
+        self._sitedens = site_density
+
+    def get_yaml(self, out):
+        super(ideal_interface, self).get_yaml(out)
+        out['site-density'] = applyUnits(self._sitedens)
+
+
+class edge(ideal_interface):
+    """A 1D boundary between two surface phases."""
+    def __init__(self, name='', elements='', species='', note='',
+                 reactions='none', site_density=0.0, phases=[], kinetics='Edge',
+                 transport='None', initial_state=None, options=[]):
+
+        ideal_interface.__init__(self, name, elements, species, note, reactions,
+            site_density, phases, kinetics, transport, initial_state, options)
+        self.thermo_model = 'edge'
+
+
+# Falloff parameterizations
+
+class Troe(object):
+    """The Troe falloff function."""
+    def __init__(self, A=0.0, T3=0.0, T1=0.0, T2=None):
+        """
+        Parameters: *A*, *T3*, *T1*, *T2*. These must be entered as pure
+        numbers with no attached dimensions.
+        """
+        self.A = A
+        self.T3 = T3
+        self.T1 = T1
+        self.T2 = T2
+
+    def get_yaml(self, out):
+        troe = FlowMap([('A', self.A), ('T3', self.T3), ('T1', self.T1)])
+        if self.T2 is not None:
+            troe['T2'] = self.T2
+        out['Troe'] = troe
+
+
+class SRI(object):
+    """ The SRI falloff function."""
+    def __init__(self, A=0.0, B=0.0, C=0.0, D=None, E=None):
+        """
+        Parameters: *A*, *B*, *C*, *D*, *E*. These must be entered as
+        pure numbers without attached dimensions.
+        """
+        self.A = A
+        self.B = B
+        self.C = C
+        self.D = D
+        self.E = E
+
+    def get_yaml(self, out):
+        sri = FlowMap([('A', self.A), ('B', self.B), ('C', self.C)])
+        if self.D is not None and self.E is not None:
```

Comment:
What if only one is `None`? Should that raise an error?

Reply:
No idea. I'm not sure I've ever seen use of this falloff function in the wild.